File: from-0401-to-0500/0442-find-all-duplicates-in-an-array/solution.go
func findDuplicates(nums []int) []int {
    var idx int
    result := []int{}
    for _, num := range nums {
        idx = abs(num) - 1
        if nums[idx] < 0 {
            result = append(result, idx + 1)
        } else {
            nums[idx] = -nums[idx]
        }
    }
    return result
}

func abs(a int) int {
    if a < 0 {
        return -a
    }
    return a
}

File: from-0401-to-0500/0442-find-all-duplicates-in-an-array/solution.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        mark = len(nums)
        for num in nums:
            nums[num%mark-1] += mark

        result = []
        for i in range(len(nums)):
            if nums[i] > 2 * mark:
                result.append(i+1)

        return result